tests/installation/ipxe_install.pm

```diff
@@ -0,0 +1,112 @@
+# SUSE's openQA tests
+#
+# Copyright © 2012-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify installation starts and is in progress
+# Maintainer: Michael Moese <mmoese@suse.de>
+
+use strict;
+use warnings;
+
+use testapi;
+use bmwqemu;
+use base "y2logsstep";
+
+use HTTP::Tiny;
+use IPC::Run;
+use Socket;
+use Time::HiRes qw(sleep);
+
+sub ipmitool {
+    my ($cmd) = @_;
+
+    my @cmd = ('ipmitool', '-I', 'lanplus', '-H', $bmwqemu::vars{IPMI_HOSTNAME}, '-U', $bmwqemu::vars{IPMI_USER}, '-P', $bmwqemu::vars{IPMI_PASSWORD});
+    push(@cmd, split(/ /, $cmd));
+
+    my ($stdin, $stdout, $stderr, $ret);
+    print @cmd;
+    $ret = IPC::Run::run(\@cmd, \$stdin, \$stdout, \$stderr);
+    chomp $stdout;
+    chomp $stderr;
+
+    die join(' ', @cmd) . ": $stderr" unless ($ret);
+    bmwqemu::diag("IPMI: $stdout");
+    return $stdout;
+}
+
+sub poweroff_host {
+    ipmitool("chassis power off");
+    while (1) {
+        sleep(3);
+        my $stdout = ipmitool('chassis power status');
+        last if $stdout =~ m/is off/;
+        ipmitool('chassis power off');
+    }
+}
+
+sub poweron_host {
+    ipmitool("chassis power on");
+    while (1) {
+        sleep(3);
+        my $stdout = ipmitool('chassis power status');
+        last if $stdout =~ m/is on/;
+        ipmitool('chassis power on');
+    }
+}
+
+sub set_bootscript {
+    my $host        = get_required_var('SUT_IP');
+    my $ip          = inet_ntoa(inet_aton($host));
+    my $http_server = get_required_var('IPXE_HTTPSERVER');
+    my $url         = "$http_server/$ip/script.ipxe";
+
+    my $autoyast = get_required_var('AUTOYAST');
+
+    my $kernel  = get_required_var('MIRROR_HTTP') . '/boot/x86_64/loader/linux';
+    my $initrd  = get_required_var('MIRROR_HTTP') . '/boot/x86_64/loader/initrd';
+    my $install = get_required_var('MIRROR_NFS');
+
+    my $bootscript = <<"END_BOOTSCRIPT";
+#!ipxe
+echo ++++++++++++++++++++++++++++++++++++++++++
+echo ++++++++++++ openQA ipxe boot ++++++++++++
+echo +    Host: $host
+echo ++++++++++++++++++++++++++++++++++++++++++
+
+kernel $kernel install=$install autoyast=$autoyast console=tty0 console=ttyS1,115200
+initrd $initrd
+boot
+END_BOOTSCRIPT
+
+    my $data     = $bootscript;
```

Comment:
it would be nice to have this script uploaded for debugging purposes. Not sure if record_info() would work here, or maybe upload_logs() with the content of this variable uploaded as a file.

Reply:
That's actually not a bad idea. I'll take care of adding this after the weekend!

Reply:
So for this purpose we don't have any files we could upload. But I added a diag with the information, so we can see it in the auotinst-log.txt. This should be sufficient I think